tools/accuracy_checker/openvino/tools/accuracy_checker/evaluators/custom_evaluators/asr_encoder_decoder_evaluator.py

```diff
@@ -273,77 +118,31 @@ def reset(self):
-        self.encoder.release()
-        self.decoder.release()
-
     def save_encoder_predictions(self):
         if self._encoder_predictions is not None:
             prediction_file = Path(self.network_info['encoder'].get('predictions', 'encoder_predictions.pickle'))
             with prediction_file.open('wb') as file:
                 pickle.dump(self._encoder_predictions, file)
 
-    def load_network(self, network_list, launcher):
-        for network_dict in network_list:
-            self._part_by_name[network_dict['name']].load_network(network_dict['model'], launcher)
-
-    def load_model(self, network_list, launcher):
-        for network_dict in network_list:
-            self._part_by_name[network_dict['name']].load_model(network_dict, launcher)
-
     def _add_raw_encoder_predictions(self, encoder_prediction):
         for key, output in encoder_prediction.items():
             if key not in self._raw_outs:
                 self._raw_outs[key] = []
             self._raw_outs[key].append(output)
 
-    def get_network(self):
-        return [{'name': 'encoder', 'model': self.encoder.network}, {'name': 'decoder', 'model': self.decoder.network}]
-
-
-class EncoderDLSDKModel(BaseModel, BaseDLSDKModel):
-    default_model_suffix = 'encoder'
-
-    def __init__(self, network_info, launcher, delayed_model_loading=False):
-        super().__init__(network_info, launcher)
-        self.input_blob, self.output_blob = None, None
-        self.with_prefix = None
-        self.is_dynamic = False
-        if not delayed_model_loading:
-            self.load_model(network_info, launcher, log=True)
 
+class EncoderDLSDKModel(BaseDLSDKModel):
     def predict(self, identifiers, input_data):
         input_data = self.fit_to_input(input_data)
         results = self.exec_network.infer(input_data)
         return results, results[self.output_blob]
 
-    def release(self):
-        del self.exec_network
-        del self.launcher
-
-    def fit_to_input(self, input_data):
-        has_info = hasattr(self.exec_network, 'input_info')
-        if has_info:
-            input_info = self.exec_network.input_info[self.input_blob].input_data
-        else:
-            input_info = self.exec_network.inputs[self.input_blob]
-        if self.input_blob in self.dynamic_inputs or tuple(input_info.shape) != np.shape(input_data):
-            self._reshape_input({self.input_blob: np.shape(input_data)})
 
-        return {self.input_blob: np.array(input_data)}
-
-
-class DecoderDLSDKModel(BaseModel, BaseDLSDKModel):
-    default_model_suffix = 'decoder'
-
-    def __init__(self, network_info, launcher, delayed_model_loading=False):
-        super().__init__(network_info, launcher)
-        self.input_blob, self.output_blob = None, None
+class DecoderDLSDKModel(BaseDLSDKModel):
+    def __init__(self, network_info, launcher, suffix=None, delayed_model_loading=False):
+        super().__init__(network_info, launcher, suffix, delayed_model_loading)
```

Comment:
please setup adapter before super init, otherwise you can get AttributeError during set_input_output

Reply:
Done